File: Monitor_matlab/Classify.m
function [ score ] = Classify(voice, fs, nBit)
%Classify Classifying Function for baby cry using 5 features extracted by corresponding functions.
%   input: voice data
%   input: sample rate
%   input: number of bits per sample
%   output: is this a baby cry? (0 for no, 1 for yes)
%   To do : construct a decision tree for classification

%     fid = fopen('result.txt','a');
    
    stdPitchFreq = 10;  %5 frames
    stdAvgFreq = 1000;
    stdSTE = 100;
    stdAcc = 200;
    stdZCR = 40;
    stdMFCC = 0;
    
    score = 0;
    
    
    feature_Freq = Freq(voice, fs, nBit);
    [feature_PitchFreq feature_PitchFreq2 ]= PitchFreq(voice , fs, nBit);
    [feature_STE feature_STEAcc ]=STE(voice, fs, nBit);
    feature_ZCR = ZCR(voice, fs, nBit);
    
    disp(feature_Freq);
    disp(feature_PitchFreq);
    disp(feature_PitchFreq2);
    disp(feature_STE);
    disp(feature_STEAcc);
    disp(feature_ZCR);
    
    if feature_Freq > stdAvgFreq
        score = score + 1;
        disp('freq yes');
    end
    
    if feature_PitchFreq > stdPitchFreq
        score = score + 1;
        disp('pitch freq yes');
    end
    
%     if feature_STE > stdSTE
%         score = score + 1;
%     end
    
    if feature_STEAcc > stdAcc
        score = score + 1;
    end
    
    if feature_ZCR > stdZCR
        score = score + 1;
        disp('zcr yes');
    end
    
    disp(score);
    
    if score >= 3
        disp('baby cry');
    end
    
    
%     fprintf(fid, '%f\t%f\t%f\t%f\t%f\t%f\n', feature_Freq,feature_PitchFreq, feature_PitchFreq2, feature_STE, feature_STEAcc, feature_ZCR);
%     
%     fclose(fid);
    
%     score = 

    
end

File: Monitor_c/Classify.c
#include <stdio.h>
#include "Freq.h"
#include "absfft.h"
#include "clipping.h"
#include "enframe.h"
#include "max.h"
#include "mean.h"
#include "medfilt3.h"
#include "PitchFreq.h"
#include "smoothing.h"
#include "STE.h"
#include "ZCR.h"
#include "zcro.h"
#include <malloc.h>

int Classify(double data[], int size, int fs, int nBit)
{
	int score = 0;
	int stdPitchFreq = 20;  %5 frames
    	double stdAvgFreq = 1000;
	double stdSTE = 30;
	double stdAcc = 200;
	int stdZCR = 40;
    
	double feature_Freq,feature_ste,feature_steacc,feature_pf2;
	int feature_pf1;
	int feature_zcr;

	feature_Freq = Freq(data,size,fs,nBit);	
	printf("feature freq finished:%lf\n",feature_Freq);

	feature_Freq = Freq(data,size,fs,nBit);
	printf("feature freq finished:%lf\n",feature_Freq);

	STE(data,size,fs,nBit,&feature_ste,&feature_steacc);
	printf("feature ste finished: %lf\t%lf\n",feature_ste,feature_steacc);

	feature_zcr = ZCR(data,size,fs,nBit);
	printf("feature zcr finished:%d\n",feature_zcr);

	PitchFreq(data,size,fs,nBit,&feature_pf1,&feature_pf2);	
	printf("feature pf finished:%d\t%lf\n",feature_pf1,feature_pf2);

	if(feature_Freq > stdAvgFreq)
	{
		score = score + 2;
		printf("freq yes");
	}
    
	if(feature_PitchFreq > stdPitchFreq)
	{
		score = score + 5;
		printf("pitch freq yes");
	}

    
	if(feature_STE > stdSTE)
	{
		score = score + 1;
		printf("ste yes");
	}
    
	if(feature_STEAcc > stdAcc)
	{
		score = score + 1;
		printf("steacc yes");
	}
    
    	if(feature_ZCR > stdZCR)
	{
        	score = score + 1;
		printf("zcr yes");
	}
    
	printf("%d\n",score);
    
	if(score >= 5)
	{
		printf("baby cry!\n");
		return 1;
	}
	else
		return 0;
}
    

